src/app/shared/helpers/map-analytics-screen-name.ts

```diff
@@ -0,0 +1,15 @@
+import { AnalyticsScreenNames } from '@providers/analytics/analytics.model';
+
+/**
+ * Map screen name from page name of component to the enum used for analytics
+ * @param sourcePage
+ */
+export function mapAnalyticsScreenName(sourcePage: string): AnalyticsScreenNames {
+  const screenName = {
+    JournalPage: AnalyticsScreenNames.JOURNAL,
+    TestCentreJournalPage: AnalyticsScreenNames.TEST_CENTRE_JOURNAL,
+    RekeySearchPage: AnalyticsScreenNames.REKEY_SEARCH,
+    default: '',
+  };
+  return screenName[sourcePage] || screenName.default;
+}
```

Comment:
Think we can make this more type conscious - would assume with this method we would need to be careful about the name of the key, whereas if we make it a Record we should be able to specify the key types i.e.

```
export function mapAnalyticsScreenName(sourcePage: string): AnalyticsScreenNames {
  const screenName = {
    JournalPage: AnalyticsScreenNames.JOURNAL,
    TestCentreJournalPage: AnalyticsScreenNames.TEST_CENTRE_JOURNAL,
    RekeySearchPage: AnalyticsScreenNames.REKEY_SEARCH,
  } as Record<PageNameKeys, AnalyticsScreenNames>;

  return screenName[sourcePage] || '';
}
``` 
You then will need to make sure that ```JournalPage```, ```TestCentreJournalPage ``` and ```RekeySearchPage``` exist in the ```PageNameKeys``` type. e.g.

```
export type PageNameKeys =
  'JournalPage' |
  'TestCentreJournalPage' |
  'RekeySearchPage' |.
....
```

Reply:
updated with suggest code improvements